tests/db-updates/class-wc-tests-update-is-read-from-last-read.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+use Automattic\WooCommerce\Admin\Notes\Note;
+
+/**
+ * DB Update test for wc_admin_update_300_update_is_read_from_last_read()
+ *
+ * @package WooCommerce\Admin\Tests\DBUpdates
+ */
+class WC_Tests_Update_Is_Read_From_Last_Read extends WC_Unit_Test_Case {
+	/**
+	 * @var object current user
+	 */
+	private $user;
+
+	/**
+	 * setUp
+	 */
+	public function setUp() {
+		parent::setUp();
+
+		$this->add_is_read_col();
+		$this->user = $this->factory->user->create(
+			array(
+				'role' => 'administrator',
+			)
+		);
+	}
+
+	/**
+	 * Adds is_read col to the notes table.
+	 */
+	protected function add_is_read_col() {
+		global $wpdb;
+		// make sure is_read col has been added successfully.
+		$notes_table_cols = $wpdb->get_results( "show columns from {$wpdb->prefix}wc_admin_notes" );
+		$notes_table_cols = array_filter(
+			$notes_table_cols,
+			function( $row ) {
+				// phpcs:ignore
+				return 'is_read' === $row->Field;
+			}
+		);
+		$has_is_read_col  = count( $notes_table_cols ) > 0;
+
+		if ( ! $has_is_read_col ) {
+			$wpdb->query(
+				"
+			alter table {$wpdb->prefix}wc_admin_notes add column is_read tinyint(1)
+		"
+			);
+		}
+	}
+
+	/**
+	 * Given woocommerce_admin_activity_panel_inbox_last_read does not exist
+	 * When the update runs
+	 * Then it should not update is_read col
+	 */
+	public function test_update_does_not_run_when_usermeta_does_not_exist() {
+		global $wpdb;
+
+		// Given.
```

Comment:
These comments can probably be removed.

Reply:
I find these gherkin language words (Given When Then) make tests more readable. I can remove it if you find it as a distraction

Reply:
I don't think they're necessarily bad, but I'd consider this given/when/then pattern pretty common for testing and not necessarily something that needs to be written each time.

Personally, I stop to read each comment because I assume the comment is explaining something that the code cannot adequately explain or we're in a rare situation that needs further explanation.  Excessive comments will slow down reading of the code, but of course can be a good thing if the function naming doesn't describe what's happening.  Better function or variable naming can often replace comments.

All that said, I'm really not strongly opinionated on this and I've spent a lot more time writing this than reading the comments. 😄

Reply:
> All that said, I'm really not strongly opinionated on this and I've spent a lot more time writing this than reading the comments.

:) I totally understand what you mean. Let's remove it. I mean..it's not like a standard we follow anyway.

Reply:
@joshuatf Removed in https://github.com/woocommerce/woocommerce-admin/pull/7945/commits/47b2cda10f9efba9d6309fbe4e60d84ef19270a1